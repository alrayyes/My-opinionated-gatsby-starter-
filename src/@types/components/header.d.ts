export interface HeaderInterface {
  siteTitle: string
}
