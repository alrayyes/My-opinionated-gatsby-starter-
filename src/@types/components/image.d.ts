import { FluidObject } from "gatsby-image"

export interface ImageDataInterface {
  placeholderImage: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
}
