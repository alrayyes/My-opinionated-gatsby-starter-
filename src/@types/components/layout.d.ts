export interface LayoutInterface {
  children: React.ReactNode
}

export interface LayoutDataInterface {
  site: {
    siteMetadata: {
      title: string
    }
  }
}
