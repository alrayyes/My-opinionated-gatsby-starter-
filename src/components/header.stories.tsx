import { storiesOf } from "@storybook/react"
import React from "react"
import Header from "./header"

storiesOf("Header component", module).add(`default`, () => (
  <Header siteTitle={"My Opinionated Gatsby Default Starter"} />
))
