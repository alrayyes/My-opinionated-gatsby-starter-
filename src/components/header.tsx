import { Link } from "gatsby"
import React from "react"
import { HeaderInterface } from "../@types/components/header"
import styled from "@emotion/styled"

const Header: React.FC<HeaderInterface> = ({ siteTitle }) => {
  const Header = styled.header`
    background: rebeccapurple;
    margin-bottom: 1.45rem;
  `

  const Container = styled.div`
    margin: 0 auto;
    max-width: 960;
    padding: 1.45rem 1.0875rem;
  `

  const H1 = styled.h1`
    margin: 0;
  `

  const StyledLink = styled(Link)`
    color: white;
    text-decoration: none;
  `

  return (
    <Header>
      <Container>
        <H1>
          <StyledLink to="/">{siteTitle}</StyledLink>
        </H1>
      </Container>
    </Header>
  )
}

export default Header
