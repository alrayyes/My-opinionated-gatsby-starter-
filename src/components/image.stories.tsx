import { storiesOf } from "@storybook/react"
import React from "react"
import Image from "./image"

storiesOf("Image component", module).add(`default`, () => <Image />)
