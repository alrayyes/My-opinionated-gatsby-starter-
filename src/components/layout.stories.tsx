import { storiesOf } from "@storybook/react"
import React from "react"
import Layout from "./layout"

storiesOf("Layout component", module).add(`default`, () => (
  <Layout>content</Layout>
))
