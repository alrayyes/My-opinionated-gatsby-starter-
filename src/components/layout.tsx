/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react"
import { useStaticQuery, graphql } from "gatsby"

import Header from "./header"
import "../utils/layout.css"
import {
  LayoutDataInterface,
  LayoutInterface,
} from "../@types/components/layout"

const Layout: React.FC<LayoutInterface> = ({ children }) => {
  const data: LayoutDataInterface = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  return (
    <>
      <Header siteTitle={data.site.siteMetadata.title} />
      <div className={"layoutContainer"}>
        <main>{children}</main>
        <footer>
          Â© {new Date().getFullYear()}, Built with{" "}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer>
      </div>
    </>
  )
}

export default Layout
