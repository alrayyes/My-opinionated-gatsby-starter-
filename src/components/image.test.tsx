import renderer from "react-test-renderer"
import React from "react"
import Image from "./image"
import * as Gatsby from "gatsby"
import { ImageDataInterface } from "../@types/components/image"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

describe("Image", () => {
  beforeEach(() => {
    const data: ImageDataInterface = {
      placeholderImage: {
        childImageSharp: {
          fluid: {
            base64:
              "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAACYklEQVQ4y42Uy28SQRjA+dM8efDmwYN6qF6qiSY+Y/WgQRMibY00TaWNBSRSCraYQtHl/bR0KyxQWCgWWAqU8izl/Sq7rLNsRHlVJpvJtzPfb77nDIOcZHSoqZSrp4+KtXIziubaLRysMCZiCYqOoVnhjNEi8RcztdxxeTzc6VBfT+5O2Vhpb+vw4wMdZ0ppWvP9xzLeJoDNThf2W+Nz1+XzNxQubSToSKKW+BDc+WOnkshhSVgeCiGpViZMEg1oxc26Knt+ae3bEtJTZwzE1kXLccG0+sOOlrcvZXvsczPkITfsa20vwIKnhsh+BnjUarT74Gb13CY7KBVJMv3z4N1NszQYsMWM62HNrCis/GxXn0iYls23uz5LPBcv0bH8hUH2XRoM85ySXv7JBtO87jMIvWq+H5GoYIHCLA1ZxD6Qap3Ak8IKfW7TJ50lK6uP9E6RgndHaODtCJ6Z5RyHfnE7j6gRbcKlCYNSt+rtETHTpUGgEP8FYmdNqd/Mo7aiVWTfuH2L9xASvfxxlqr01EYkrJszvNkgW9bH0OuFr+99m+y9IOeyU6zIp/Hubp/yMEztlzFPwOhdvq+nIoS1JNn4t2sugCmVsDvPe2KKolnZLCxhOcAKQRDDXTQaVi46lqYhIBwHTrl3oWqhMRDtaJge37lOBMKo4tfbqhVX0J7snTsWps8uZWuoOQY6CcjpSIF55UvmqNgr5wUwtV1IVdnXtnSfPEB2qjDNqnvczRl0m+j6Jn5lXb6nAQJqinmN0ZEBj03YLzghY8PnTRz80o/GRJZpOLCb0PM9BN7pvUEjx28V00WUg9jIVwAAAABJRU5ErkJggg==",
            aspectRatio: 1,
            src:
              "/static/6d91c86c0fde632ba4cd01062fd9ccfa/ee604/gatsby-astronaut.png",
            srcSet:
              "/static/6d91c86c0fde632ba4cd01062fd9ccfa/69585/gatsby-astronaut.png 200w,\n/static/6d91c86c0fde632ba4cd01062fd9ccfa/497c6/gatsby-astronaut.png 400w,\n/static/6d91c86c0fde632ba4cd01062fd9ccfa/ee604/gatsby-astronaut.png 800w",
            srcSetWebp:
              "/static/6d91c86c0fde632ba4cd01062fd9ccfa/61e93/gatsby-astronaut.webp 200w,\n/static/6d91c86c0fde632ba4cd01062fd9ccfa/1f5c5/gatsby-astronaut.webp 400w,\n/static/6d91c86c0fde632ba4cd01062fd9ccfa/58556/gatsby-astronaut.webp 800w",
            srcWebp:
              "/static/6d91c86c0fde632ba4cd01062fd9ccfa/58556/gatsby-astronaut.webp",
            sizes: "(max-width: 800px) 100vw, 800px",
          },
        },
      },
    }

    useStaticQuery.mockImplementation(() => data)
  })

  it("renders correctly", () => {
    const tree = renderer.create(<Image />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
