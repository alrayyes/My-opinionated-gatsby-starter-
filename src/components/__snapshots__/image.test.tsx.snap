// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  <img
    alt=""
    aria-hidden={true}
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAACYklEQVQ4y42Uy28SQRjA+dM8efDmwYN6qF6qiSY+Y/WgQRMibY00TaWNBSRSCraYQtHl/bR0KyxQWCgWWAqU8izl/Sq7rLNsRHlVJpvJtzPfb77nDIOcZHSoqZSrp4+KtXIziubaLRysMCZiCYqOoVnhjNEi8RcztdxxeTzc6VBfT+5O2Vhpb+vw4wMdZ0ppWvP9xzLeJoDNThf2W+Nz1+XzNxQubSToSKKW+BDc+WOnkshhSVgeCiGpViZMEg1oxc26Knt+ae3bEtJTZwzE1kXLccG0+sOOlrcvZXvsczPkITfsa20vwIKnhsh+BnjUarT74Gb13CY7KBVJMv3z4N1NszQYsMWM62HNrCis/GxXn0iYls23uz5LPBcv0bH8hUH2XRoM85ySXv7JBtO87jMIvWq+H5GoYIHCLA1ZxD6Qap3Ak8IKfW7TJ50lK6uP9E6RgndHaODtCJ6Z5RyHfnE7j6gRbcKlCYNSt+rtETHTpUGgEP8FYmdNqd/Mo7aiVWTfuH2L9xASvfxxlqr01EYkrJszvNkgW9bH0OuFr+99m+y9IOeyU6zIp/Hubp/yMEztlzFPwOhdvq+nIoS1JNn4t2sugCmVsDvPe2KKolnZLCxhOcAKQRDDXTQaVi46lqYhIBwHTrl3oWqhMRDtaJge37lOBMKo4tfbqhVX0J7snTsWps8uZWuoOQY6CcjpSIF55UvmqNgr5wUwtV1IVdnXtnSfPEB2qjDNqnvczRl0m+j6Jn5lXb6nAQJqinmN0ZEBj03YLzghY8PnTRz80o/GRJZpOLCb0PM9BN7pvUEjx28V00WUg9jIVwAAAABJRU5ErkJggg=="
    style={
      Object {
        "height": "100%",
        "left": 0,
        "objectFit": "cover",
        "objectPosition": "center",
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "transitionDelay": "500ms",
        "width": "100%",
      }
    }
  />
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source type='image/webp' srcset=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/61e93/gatsby-astronaut.webp 200w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/1f5c5/gatsby-astronaut.webp 400w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/58556/gatsby-astronaut.webp 800w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><source srcset=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/69585/gatsby-astronaut.png 200w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/497c6/gatsby-astronaut.png 400w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/ee604/gatsby-astronaut.png 800w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/69585/gatsby-astronaut.png 200w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/497c6/gatsby-astronaut.png 400w,
/static/6d91c86c0fde632ba4cd01062fd9ccfa/ee604/gatsby-astronaut.png 800w\\" src=\\"/static/6d91c86c0fde632ba4cd01062fd9ccfa/ee604/gatsby-astronaut.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
